File: BOOKWORM_PRO/Bookworm/src/main/java/com/example/Controller/BeneficiaryController.java
package com.example.Controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import com.example.Services.BeneficiaryService;
import com.example.models.Beneficiary;

@RestController
@RequestMapping("/api/beneficiaries")
@CrossOrigin(origins = "*")
public class BeneficiaryController {

    @Autowired
    private BeneficiaryService beneficiaryService;

    @PostMapping
    public Beneficiary createBeneficiary(@RequestBody Beneficiary beneficiary) {
        return beneficiaryService.createBeneficiary(beneficiary);
    }

    @PutMapping("/{id}")
    public Beneficiary updateBeneficiary(@PathVariable int id,
                                         @RequestBody Beneficiary beneficiary) {
        return beneficiaryService.updateBeneficiary(id, beneficiary);
    }

    @GetMapping("/{id}")
    public Beneficiary getBeneficiaryById(@PathVariable int id) {
        return beneficiaryService.getBeneficiaryById(id);
    }

    @GetMapping
    public List<Beneficiary> getAllBeneficiaries() {
        return beneficiaryService.getAllBeneficiaries();
    }

    @DeleteMapping("/{id}")
    public String deleteBeneficiary(@PathVariable int id) {
        beneficiaryService.deleteBeneficiary(id);
        return "Beneficiary deleted successfully with id: " + id;
    }
}

File: BookWorm-NET/BookWormNET/Controllers/BeneficiaryController.cs



using BookWormNET.Models;
using BookWormNET.Services;
using BookWormNET.Services.Interfaces;
using Microsoft.AspNetCore.Mvc;

namespace BookWormNET.Controllers
{
    [ApiController]
    [Route("api/beneficiaries")]
    public class BeneficiariesController : ControllerBase
    {
        private readonly IBeneficiaryService _beneficiaryService;

        // Constructor Injection (instead of @Autowired)
        public BeneficiariesController(IBeneficiaryService beneficiaryService)
        {
            _beneficiaryService = beneficiaryService;
        }

        // POST: api/beneficiaries
        [HttpPost]
        public ActionResult<Beneficiary> CreateBeneficiary([FromBody] Beneficiary beneficiary)
        {
            var result = _beneficiaryService.CreateBeneficiary(beneficiary);
            return Ok(result);
        }

        // PUT: api/beneficiaries/{id}
        [HttpPut("{id}")]
        public ActionResult<Beneficiary> UpdateBeneficiary(int id, [FromBody] Beneficiary beneficiary)
        {
            var result = _beneficiaryService.UpdateBeneficiary(id, beneficiary);
            return Ok(result);
        }

        // GET: api/beneficiaries/{id}
        [HttpGet("{id}")]
        public ActionResult<Beneficiary> GetBeneficiaryById(int id)
        {
            var beneficiary = _beneficiaryService.GetBeneficiaryById(id);
            if (beneficiary == null)
                return NotFound();

            return Ok(beneficiary);
        }

        // GET: api/beneficiaries
        [HttpGet]
        public ActionResult<IEnumerable<Beneficiary>> GetAllBeneficiaries()
        {
            return Ok(_beneficiaryService.GetAllBeneficiaries());
        }

        // DELETE: api/beneficiaries/{id}
        [HttpDelete("{id}")]
        public IActionResult DeleteBeneficiary(int id)
        {
            _beneficiaryService.DeleteBeneficiary(id);
            return Ok($"Beneficiary deleted successfully with id: {id}");
        }
    }
}
